### PR TITLE
Allow user to customize x-axis (months survival) in survival plot

### PR DESCRIPTION
**Pull request by dippindots**
Fix cBioPortal/cbioportal#2176

- Provide a range slider and text input to configure the x-axis of the survival chart
image

### CODE REVIEW

File: src/pages/resultsView/survival/SurvivalChart.tsx [hunk 1]
```diff
@@ -410,6 +414,24 @@ export default class SurvivalChart extends React.Component<ISurvivalChartProps,
...
 
+    @computed get maximumDataMonthValue() {
+        return _.chain(this.sortedGroupedSurvivals)
+                .map((survivals) => survivals[survivals.length - 1].months)
+                .max()
+                .ceil()
+                .value();
+    }
+
+    @autobind
```
Comment (alisman) [inline src/pages/resultsView/survival/SurvivalChart.tsx]: @action

File: src/pages/resultsView/survival/SurvivalChart.tsx [hunk 2]
```diff
@@ -410,6 +414,24 @@ export default class SurvivalChart extends React.Component<ISurvivalChartProps,
...
+    onSliderChange(value: number) {
+        this.sliderValue = value;
+    }
+
+    @autobind
```
Comment (alisman) [inline src/pages/resultsView/survival/SurvivalChart.tsx]: @action

**Comment by alisman** [comment]
@dippindots nice work.  lets add the unit (months surivval) after text box.  then please send a link to Niki for final test

**Comment by dippindots** [comment]
@alisman Sure, I will update soon. Thanks for the review!

**Pull request merged**